packages/react-hooks/src/hooks/media.ts

```diff
@@ -0,0 +1,35 @@
+import {useState, useEffect} from 'react';
+
+export function useMedia<ValueType>(
+  queries: string[],
+  values: ValueType[],
+  defaultValue: ValueType,
+) {
+  const mediaQueryLists = queries.map(query => {
+    return window.matchMedia(query);
+  });
+
+  const getValue = () => {
+    const index = mediaQueryLists.findIndex(mediaQueryList => {
+      return mediaQueryList.matches;
+    });
+    return typeof values[index] === 'undefined' ? defaultValue : values[index];
+  };
+
+  const [value, setValue] = useState(getValue);
+
+  useEffect(() => {
+    const handler = () => {
+      setValue(getValue);
+    };
+    mediaQueryLists.forEach(mediaQueryList => {
+      mediaQueryList.addListener(handler);
+    });
+    return () =>
+      mediaQueryLists.forEach(mediaQueryList => {
+        mediaQueryList.removeListener(handler);
+      });
+  }, []);
```

Comment:
Why does the linter does not like having an empty dependency array? What I want to simulate `componentDidMount` and `componentWillUnmount` .

Reply:
You have to disabled the rule with:
`// eslint-disable-next-line react-hooks/exhaustive-deps`